### PR TITLE
Validation & Password recovery integration

### PR DESCRIPTION
**Pull request by joaopcartier**

### CODE REVIEW

File: frontend/src/components/common/modal/style.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const modalCustomStyles: ReactModal.Styles = {
   overlay: {
     backgroundColor: "rgba(0,0,0,0.5)",
     display: "flex",
-
+    overflowY: "scroll",
```
Comment (joaopcartier) [inline frontend/src/components/common/modal/style.ts:16]: In order to scroll the modal when page's height is hiding it

File: frontend/src/components/group/GroupDetails.tsx [hunk 2]
```diff
@@ -23,10 +23,6 @@ export default function GroupDetails() {
 
   const dispatch = useAppDispatch();
 
-  useEffect(() => {
-    onLoad(Number(id));
-  }, [id]);
-
```
Comment (joaopcartier) [inline frontend/src/components/group/GroupDetails.tsx:29]: We were not using it to retrieve data, resulting in a unnecessary API call

File: frontend/src/components/group/GroupDetails.tsx [hunk 3]
```diff
@@ -81,11 +77,3 @@ export default function GroupDetails() {
...
   );
 }
-
-async function onLoad(id: number) {
-  try {
-    await store.dispatch(requestGroup(id));
-  } catch (err) {
-    console.log("Error loading group", err);
-  }
-}
```
Comment (joaopcartier) [inline frontend/src/components/group/GroupDetails.tsx:91]: We were not using it to retrieve data, resulting in a unnecessary API call

File: frontend/src/components/role/RoleDetails.tsx [hunk 4]
```diff
@@ -27,8 +27,8 @@ export default function RoleDetails() {
   const dispatch = useAppDispatch();
 
   useEffect(() => {
-    onLoad(Number(id));
-  }, [id]);
```
Comment (joaopcartier) [inline frontend/src/components/role/RoleDetails.tsx:31]: We were not using it to retrieve data, resulting in a unnecessary API call

File: frontend/src/components/role/RoleDetails.tsx [hunk 5]
```diff
@@ -27,8 +27,8 @@ export default function RoleDetails() {
...
+    onLoad();
+  }, []);
```
Comment (joaopcartier) [inline frontend/src/components/role/RoleDetails.tsx:31]: We don't need to make a request to get the user.
Only a request to update the groups from the backend in order to pick when editing a Role if we want to change the group

File: frontend/src/components/user/UserDetails.tsx [hunk 6]
```diff
@@ -31,10 +31,6 @@ export default function UserDetails() {
 
   const dispatch = useAppDispatch();
 
-  useEffect(() => {
-    onLoad(Number(id));
-  }, [id]);
-
```
Comment (joaopcartier) [inline frontend/src/components/user/UserDetails.tsx:37]: We were not using it to retrieve data, resulting in a unnecessary API call

File: frontend/src/components/user/UserDetails.tsx [hunk 7]
```diff
@@ -140,11 +136,3 @@ export default function UserDetails() {
...
   );
 }
-
-async function onLoad(id: number) {
-  try {
-    await store.dispatch(requestUser(id));
-  } catch (err) {
-    console.log("Error loading user list", err);
-  }
-}
```
Comment (joaopcartier) [inline frontend/src/components/user/UserDetails.tsx:150]: We were not using it to retrieve data, resulting in a unnecessary API call

File: frontend/src/routes.tsx [hunk 8]
```diff
@@ -39,6 +36,8 @@ export default function Routes() {
             </Route>
           </>
         </ProtectedRoute>
+        <PasswordReset />
```
Comment (joaopcartier) [inline frontend/src/routes.tsx:39]: Moved here due to route hierarchy

**Review by anibalgatica — APPROVED** [review]

**Pull request merged**